client.go

```diff
@@ -304,9 +315,64 @@ func (client *client) GetOffset(topic string, partitionID int32, time int64) (in
 	return offset, err
 }
 
+func (client *client) Coordinator(consumerGroup string) (*Broker, error) {
+	client.lock.RLock()
+	coordinatorID, coordinatorCached := client.coordinators[consumerGroup]
+	client.lock.RUnlock()
+
+	if !coordinatorCached {
+		var err error
+		err = client.RefreshCoordinator(consumerGroup)
+		if err != nil {
+			return nil, err
+		}
+
+		client.lock.RLock()
+		coordinatorID = client.coordinators[consumerGroup]
+		client.lock.RUnlock()
+	}
+
+	client.lock.RLock()
+	coordinator := client.brokers[coordinatorID]
```

Comment:
Because you've released and re-taken the lock this isn't guaranteed to be present. It may be necessary to have a `cachedCoordinator` method like `cachedLeader` that reads the ID and the broker under a single lock, and handles a missing broker by returning the appropriate error instead of `nil, nil`.